### DISCUSSION TITLE
Choosing what to compile

### DISCUSSION THREAD

**Discussion by Nardol** [BearWare/TeamTalk5#1495] [General]
Hello,

As I have no answer from you by mail @bear101 I ask my question here :slightly_smiling_face:

I would like to only compile parts of TeamTalk5 I need for testing from master branch.

I read Library/TeamTalkLib/README.md but without real success.
I don't want to compile TeamTalk pro, documentations, TeamTalk Server nor anything Java related.
In fact I only need TeamTalk5 DLL and the QT client.
I compile on Debian Buster and have read Ubuntu 18 Github workflows to be sure of the command line.

For the moment, I think I am near the desired result with this very long line:
`make CMAKE_EXTRA="-DTOOLCHAIN_CATCH2=OFF -DBUILD_TEAMTALK_LIB=OFF -DBUILD_TEAMTALK_PROLIB=OFF -DBUILD_TEAMTALK_PRODLL=OFF -DBUILD_TEAMTALK_SERVERS=OFF -DBUILD_TEAMTALK_PROSERVER_EXAMPLE=OFF -DBUILD_TEAMTALK_SRVEXE=OFF -DBUILD_TEAMTALK_PROSRVEXE=OFF -DBUILD_TEAMTALK_JTEAMTALKSPAMBOT=OFF -DBUILD_TEAMTALK_JTEAMTALK=OFF -DBUILD_TEAMTALK_JAR=OFF -DBUILD_TEAMTALK_PROJAR=OFF -DBUILD_TEAMTALK_CLIENT_LOG_EXAMPLE=OFF -DBUILD_TEAMTALK_PROCLIENT_LOG_EXAMPLE=OFF -DBUILD_TEAMTALK_PROCLIENT_QT_EXAMPLE=OFF -DBUILD_TEAMTALK_JNI=OFF -DBUILD_TEAMTALK_PROJNI=OFF -DBUILD_TEAMTALK_CLIENT_DOCS=OFF -DBUILD_TEAMTALK_SDK_DOCS=OFF" -C Build deb64`

Before doing it I type `source env.sh`

But it looks like it still compiles some documentation.
And without the examples switched to off, TeamTalk pro tries to be compiled which throws an error because for the lib, the result is the desired one

What do I miss?

And last question: can this command line be optimized (shortened)?

**Comment by bear101** [comment]
I suggest you cmake directly:
1. `cmake -S /home/foo/TeamTalk5 -B /home/foo/build -DBUILD_TEAMTALK_CORE=ON`
2. `cmake --build /home/foo/build --target TeamTalk5`
